include/xtensor/xrandom.hpp

```diff
@@ -261,7 +261,16 @@ namespace xt
 
             if (de.dimension() == 1)
             {
-                std::shuffle(de.storage().begin(), de.storage().end(), engine);
+                auto first = de.storage().begin();
+                auto last = de.storage().end();
+
+                for (auto i = (last - first) - 1; i > 0; --i)
+                {
+                    std::uniform_int_distribution<decltype(i)> dist(0, i);
```

Comment:
Note that `decltype(i)` will probably be `std::ptrdiff_t` (the `difference_type`) while we set it to `size_type` below.

One of them should probably be adjusted.